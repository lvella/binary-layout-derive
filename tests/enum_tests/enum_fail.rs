use binary_layout_derive::LayoutAsEnumRepr;

#[derive(LayoutAsEnumRepr)]
#[repr(u32)]
enum MyBadEnum {
    Variant1 = 1,
}

// Missing Into<u32> and TryFrom<u32> implementations, should fail.

fn main() {}
