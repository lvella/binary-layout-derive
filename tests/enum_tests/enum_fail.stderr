error[E0277]: the trait bound `MyBadEnum: From<u32>` is not satisfied
 --> tests/enum_tests/enum_fail.rs:3:10
  |
3 | #[derive(LayoutAsEnumRepr)]
  |          ^^^^^^^^^^^^^^^^ the trait `From<u32>` is not implemented for `MyBadEnum`
  |
  = note: required for `u32` to implement `Into<MyBadEnum>`
  = note: required for `MyBadEnum` to implement `TryFrom<u32>`
  = note: this error originates in the derive macro `LayoutAsEnumRepr` (in Nightly builds, run with -Z macro-backtrace for more info)
